### PR TITLE
fixes#2756 close keyboard when clicked on menu buttons of download screen and choose language

### LINKED ISSUE #2756: Keyboard doesnt hide when clicked on menu buttons of download screen and choose language
**Describe the bug**

The keyboard doesn't hide even when the editText is not focused
1) when clicked on ``choose a language `` from the download screen keyboard doesn't close and stays open in the select language screen
vice versa 
from the select languages screen when clicked on the ``tick``  icon keyboard doesn't close

**Expected behavior**
the keyboard should hide once editText has no focus
**Steps to reproduce the behavior:**
1. go to the download screen, click on search, see the keyboard open
2. now click on 
image
and see the keyboard still open on a different screen
1. go to select languages screen and click on search see the keyboard open
4. now click on tick and see the keyboard open on the download screen
-->

sorry if I couldn't explain properly
see this screen recording this might help in understanding

**Screenshots**

https://user-images.githubusercontent.com/75211982/144816520-36d0d53b-6c51-4dc1-bad8-dfc25f6ea329.mp4

**Environment**
- Version of Kiwix Android: 3.4.5
- Device: realme 6/ emulator
- OS version: android10/11

I would like to work on this issue :)

### PR DESCRIPTION
**Pull request by cdhiraj40**
fixes #2756

Screenshots

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ fun Fragment.toast(stringId: Int, length: Int = Toast.LENGTH_LONG) {
 fun Fragment.closeKeyboard() {
   val inputMethodManager =
     requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-  inputMethodManager.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt:42]: This is here for the toggle input keyboard. We shouldn't change that.

File: app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt [hunk 2]
```diff
@@ -169,6 +169,7 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     when (item.itemId) {
       R.id.select_language -> requireActivity().navigate(R.id.languageFragment)
     }
+    closeKeyboard()
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:172]: We shouldn't add there. It should add under `select_language` id's condition.

File: core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt (hunk 1 above)
Comment (cdhiraj40) [reply core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt]: sorry for that, I removed that change.

**Review by gouri-panda — APPROVED** [review]

**Comment by cdhiraj40** [comment]
@kelson42, please merge :)

**Pull request merged**